### DISCUSSION TITLE
GroupTag Function

### DISCUSSION THREAD

**Discussion by andadina** [okieselbach/Autopilot-Manager#21] [Q&A]
Hello

Could you please share with us the grouptag function?

Thank you

**Comment by okieselbach** [comment]
Hi,

This is the actual example from Microsoft. I used the Azure Function HTTP Trigger PS example below and modified it just a little bit:

https://learn.microsoft.com/de-de/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=python-v2%2Cin-process%2Cfunctionsv2&pivots=programming-language-powershell

```
using namespace System.Net

# Input bindings are passed in via param block.
param($Request, $TriggerMetadata)

# Write to the Azure Functions log stream.
Write-Host "PowerShell HTTP trigger function processed a request."

# Interact with query parameters or the body of the request.
$name = $Request.Query.Name
if (-not $name) {
    $name = $Request.Body.Name
}

$body = "This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response."

if ($name) {
    $body = "Hello, $name. This HTTP triggered function executed successfully."
}

# Associate values to output bindings by calling 'Push-OutputBinding'.
Push-OutputBinding -Name Response -Value ([HttpResponseContext]@{
    StatusCode = [HttpStatusCode]::OK
    Body       = $body
})
```

**Reply by andadina** [reply]
Thank you for your reply. I used the actual example from Microsoft. But the created function does not work with the Autopilot Manager tool. Maybe you have an idea what could be wrong.

I registered the Autopilot Manager application in the tenant and it worked, registration was done. Then I tried the automation from your post: [Autopilot Manager with additional automation – Modern IT – Cloud – Workplace (oliverkieselbach.com)](https://oliverkieselbach.com/2023/01/24/autopilot-manager-with-additional-automations/comment-page-1/?unapproved=35910&moderation-hash=e8f17a2c80b643932d4d27c80fb139cb#comment-35910)

I’ve created the Function app and a validate function (nothing special, just the example form Microsoft). After the function was created I enabled the new function with Autopilot Manager, in Azure App Service Configuration. Exactly as you mentioned:
AppConfig:AutopilotManagerConfig:AzureFunctionsUrl set to your Azure Function URL (e.g. https://apm-functions-xxx.azurewebsites.net/api/)
AppConfig:AutopilotManagerConfig:UseValidationFunction set to true
AppConfig:AutopilotManagerConfig:ValidationFunctionKey set to "your azure function key"

When I use the app to register a new device I get the red window error, your device is not allowed to be provisioned. It looks like the Autopilot Manager does not trigger the function up, or does not reach the function. If I check the Monitor for the function no new record is visible, so no execution.
Do you have any idea what is missing, where to check what is wrong? I guess is something related to permissions….

Any hint would be highly appreciated.
Thank you

**Comment by okieselbach** [comment]
If you look at the App Service, there is the App Service logs section. Here you need to configure the logging level to Information and then have a look into the Log stream. This will show log entries from the App Service core logic and will have entries like, can't reach function, permission is incorrect etc. 
In addition to the log stream (which is a live log view), you can open the Advanced tools on the App service. This is a different portal. There you can start a cmd or powershell and get access to the filesystem of the app service. There you can download the application log files and look through them via an editor.
